test/cluster/storage/conftest.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/python3
+#
+# Copyright (C) 2025-present ScyllaDB
+#
+# SPDX-License-Identifier: LicenseRef-ScyllaDB-Source-Available-1.0
+#
+import pytest
+import pathlib
+import shutil
+import subprocess
+import uuid
+
+from typing import Callable
+from contextlib import asynccontextmanager, contextmanager
+
+from test.pylib.manager_client import ManagerClient
+from test.cluster.conftest import PHASE_REPORT_KEY
+
+
+@pytest.fixture(scope="function")
+def volumes_factory(pytestconfig, build_mode, request):
+    hash = str(uuid.uuid4())
+    base = pathlib.Path(f"{pytestconfig.getoption("tmpdir")}/{build_mode}/volumes/{hash}")
+    volumes = []
+
+    @contextmanager
+    def wrapper(sizes: list[str]):
+        try:
+            for id, size in enumerate(sizes):
+                path = base / f"scylla-{id}"
+                path.mkdir(parents=True)
+
+                subprocess.run(["sudo", "mount", "-o", f"size={size}", "-t", "tmpfs", "tmpfs", path], check=True)
+                volumes.append(path)
+            yield volumes
+        finally:
+            pass
+    yield wrapper
+
+    # The test in the storage module use unshare -mr as the launcher to be able to mount volumes.
+    # This means that the entire content of the mount is only visible inside the namespace. To keep
+    # the files on a test failure, we need to copy them out of the namespace.
+    #
+    # Copying cannot be done in the finally clause of the wrapper above as at that point test is not
+    # yet marked as failed. So the copy has to be done here.
+    #
+    # Note that since volumes are created within unshare, they will be automatically unmounted and
+    # files will be deleted so no additional cleanup is needed.
+    report = request.node.stash[PHASE_REPORT_KEY]
+    test_failed = report.when == "call" and report.failed
+    preserve_data = test_failed or request.config.getoption("save_log_on_success")
+
+    if preserve_data:
+        for id, path in enumerate(volumes):
+            shutil.copytree(path, base.parent.parent / f"scylla-{hash}-{id}")
```

Comment:
I hope that none of these tests fail because I don't want archives of filler sstables clogging up the reports.

Reply:
It follows the exact same behavior of `test.py` for the remaining tests. 

But this is actually a good point. In my tests, in order to reach the critical disk utilization, I create random blob files of a certain size that might not be deleted in case of a test failure. I sent a new version that fixes that. Now I use 
```
with random_content_file(...):
    <code>
```
so the file is always removed no matter of the test result. Now when a test fails, we copy around a few MB of data only.